lib/network_interface/worker.ex

```diff
@@ -13,45 +13,105 @@
 # limitations under the License.
 
 defmodule Nerves.NetworkInterface.Worker do
+  @moduledoc """
+  Working for NetworkInterface. 
+  See `Nerves.NetworkInterface` for more details.
+  """
+
   use GenServer
   require Logger
 
-  @moduledoc false
+  defstruct port: nil
+
+  @typedoc "State of the GenServer"
+  @type state :: %__MODULE__{port: port}
 
-  defstruct port: nil,
-            requests: []
+  @typedoc "Setup options."
+  @type options :: map 
 
+  @typedoc "Interface name"
+  @type ifname :: binary
```

Comment:
I often find that people start to get confused about `binary` and think they should be passing something in `<<>>`. I find that using `String.t` seems to be more clarifying to many people.